magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala

```diff
@@ -242,6 +265,20 @@ trait CloudFormation extends AWS {
     client.describeStackEvents(
       new DescribeStackEventsRequest().withStackName(name)
     )
+
+  def findStackByTags(tags: Map[String, List[String]], reporter: DeployReporter)(implicit keyRing:KeyRing) = {
+    val cfnStacks = client.describeStacks(
+      new DescribeStacksRequest()
+    ).getStacks.filter { stack =>
```

Comment:
`describeStacks` is a paged API so a single call will potentially only return a subset of stacks. You probably want to make this call recursive depending on the value of `getNextToken`: http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/cloudformation/model/DescribeStacksResult.html#getNextToken--